odoo/addons/base/models/ir_mail_server.py

```diff
@@ -420,24 +437,20 @@ def _get_default_from_address(self):
             return "%s@%s" % (email_from, domain)
         return tools.config.get("email_from")
 
-    def _prepare_email_message(self, message):
+    def _prepare_email_message(self, message, smtp_session):
         # Use the default bounce address **only if** no Return-Path was
         # provided by caller.  Caller may be using Variable Envelope Return
         # Path (VERP) to detect no-longer valid email addresses.
-        smtp_from = message['Return-Path'] or self._get_default_bounce_address() or message['From']
+        bounce_address = message['Return-Path'] or self._get_default_bounce_address() or message['From']
```

Comment:
Not sure to understand this from / bounce mismatch. Keeping bounces support is important.

Reply:
Now if the from_filter is set, we will use the bounce address only if the domain  of the bounce address match this from_filter. So in most cases we will use it.

But if someone has a mail server for the domain "odoo.com" and the bounce address is in the domain "asbl.com" it will not use it because otherwise the email will be considered as spam... If we do not do that, it means that we will always use the bounce address, and so if it's miss configured (the case when we do not use the bounce address the the current code) we will spoof.